dss/stepfunctions/s3copyclient/implementation.py

```diff
@@ -181,6 +184,15 @@ def calculate_range_for_part(self, part_id) -> typing.Tuple[int, int]:
         return result
 
 
+def _determine_cache_tagging(cache: bool, destination_bucket: str, destination_key: str):
+    if not cache and is_dss_bucket(destination_bucket):
+        tag_set = {"TagSet": [{"Key": "uncached", "Value": "True"}]}
+        try:
+            s3.put_object_tagging(Bucket=destination_bucket, Key=destination_key, Tagging=tag_set)
+        except ClientError as ex:
+            raise ValueError(f"Could not find s3://{destination_bucket}/{destination_key}") from ex
```

Comment:
Raising an exception here may have unexpected side effects such as triggering stepfunction retry logic.

Consider logging an error and allowing the stepfunction execution to proceed and possibly fail as usual.